value/list.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package value
+
+// List represents a list object.
+type List interface {
+	// Length returns how many items can be found in the map.
+	Length() int
+	// At returns the item at the given position in the map. It will
+	// panic if the index is out of range.
+	At(int) Value
+}
+
+// Equals compares two lists lexically.
+func ListEquals(lhs, rhs List) bool {
```

Comment:
Idea for discussion (for followup or ignore):

```
type ListOp struct {
  LHS, RHS List
}

func (o ListOp) Equals() bool { ... }
func (o ListOp) Less() bool { ... }
func (o ListOp) Compare() int { ... }

```

Not claiming this is better.

Reply:
I really don't see the point.